pkg/addon/parameters.go

```diff
@@ -43,26 +44,31 @@ func GetParameterByInstallation(ctx context.Context, client k8sclient.Client, in
 	)
 }
 
-// GetParameterByInstallType retrieves the value for an addon parameter by
-// selecting the addon name for installationType
+// GetParameterByInstallType retrieves the value for an addon parameter from
+// the addon parameters secret
 func GetParameterByInstallType(ctx context.Context, client k8sclient.Client, installationType integreatlyv1alpha1.InstallationType, namespace, parameter string) ([]byte, bool, error) {
-	addonName := GetName(installationType)
-	secretName := fmt.Sprintf("addon-%s-parameters", addonName)
-
-	secret := &corev1.Secret{}
-	if err := client.Get(ctx, k8sclient.ObjectKey{
-		Name:      secretName,
-		Namespace: namespace,
-	}, secret); err != nil {
+	secrets := &corev1.SecretList{}
+	opts := []k8sclient.ListOption{
+		k8sclient.InNamespace(namespace),
+	}
+	if err := client.List(ctx, secrets, opts...); err != nil {
 		if errors.IsNotFound(err) {
 			return nil, false, nil
 		}
-
 		return nil, false, fmt.Errorf("failed to retrieve parameters secret: %v", err)
 	}
 
-	value, ok := secret.Data[parameter]
-	return value, ok, nil
+	for _, secret := range secrets.Items{
+		matched, err := regexp.MatchString("^addon-.*-parameters$", secret.Name)
+		if err != nil{
+			return nil, false, err
```

Comment:
As we spoke about. The getting of the secret should be in its own function call. This function can be used with this function and replace the code [here](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/custom-smtp/custom-smtp.go#L34-L41)